gradient_descent.js

```diff
@@ -225,8 +278,68 @@ class Adam extends AlgorithmFirstOrder{
 
   reinitialize() {
     this.path = [];
+    this.x = this.x_ini.map(x => x);
     this.currentGradientAverage = this.x_ini.map(x => 0);
     this.currentSquareGradientAverage = this.x_ini.map(x => 0);
     this.nStep = 0;
   }
 }
+
+/** Gradient Descent with nesterov momentum must do the following task
```

Comment:
Be careful to replace the class names in the comments 😉 
```suggestion
/** BFGS implements the following interface
```